src/librustc_mir/interpret/memory.rs

```diff
@@ -73,26 +73,26 @@ impl MemoryPointer {
         MemoryPointer { alloc_id, offset }
     }
 
-    pub fn wrapping_signed_offset<'tcx>(self, i: i64, layout: &TargetDataLayout) -> Self {
-        MemoryPointer::new(self.alloc_id, value::wrapping_signed_offset(self.offset, i, layout))
+    pub(crate) fn wrapping_signed_offset<'a, L: HasDataLayout<'a>>(self, i: i64, l: L) -> Self {
```

Comment:
Ohhh I know why I'm confused, you're not using the tools in `ty::layout`!

E.g. https://github.com/rust-lang/rust/blob/ef19bd1a339212f77160f12af6d5d85d8bb36ea4/src/librustc/ty/layout.rs#L536-L537